### PR TITLE
Update README for acrn-unit-test

### PR DESCRIPTION
**Pull request by shenfang2019**
Signed-off-by: Shen Fangfang <fangfang.shen@intel.com>

### CODE REVIEW

File: README [hunk 1]
```diff
@@ -1,3 +1,55 @@
-This is the unit-test for ACRN project. We have two type of unit-test:
- - guest side. The unit-test will be run as a ACRN guest
- - host side. The uni-test will be run at ACRN host side.
+# Readme
+  This is the unit-test for ACRN project. The unit-test will run as a ACRN guest.
```
Comment (junjiemao1) [inline README]:
```suggestion
  This is the unit-test for ACRN project. The unit-test will run in an ACRN VM.
```

I assume we'll be able to exercise both pre-launched and post-launched VMs soon. so not mentioning if the `VM` is pre-launched or post-launched here.

File: README [hunk 2]
```diff
@@ -1,3 +1,55 @@
...
+  Notes: acrn-unit-test now is still test framework ready only, no much detailed test case yet.
```
Comment (junjiemao1) [inline README]:
```suggestion
  Notes: acrn-unit-test is ready to use as a unit test framework but there is not many detailed test cases yet. 
```

File: README [hunk 3]
```diff
@@ -1,3 +1,55 @@
-This is the unit-test for ACRN project. We have two type of unit-test:
- - guest side. The unit-test will be run as a ACRN guest
- - host side. The uni-test will be run at ACRN host side.
+# Readme
+  This is the unit-test for ACRN project. The unit-test will run as a ACRN guest.
+  Notes: acrn-unit-test now is still test framework ready only, no much detailed test case yet. 
+         But you can try and add new cases according to the following descriptions.
```
Comment (junjiemao1) [inline README]:
```suggestion
         But you can try and add new cases following the steps below.
```

Better put a reference to the last section in this file.

File: README [hunk 4]
```diff
@@ -1,3 +1,55 @@
...
+# acrn-unit-test directory structure.
+    ./guest:			configure script, top-level Makefile
+    ./guest/scripts:		helper scripts for building and running tests
+    ./guest/lib:		general architecture neutral services for the tests
+    ./guest/lib/x86:		architecture dependent services for the tests
+    ./guest/x86:		the sources of the tests and the created objects/images
+
+# acrn-unit-test building steps.
+  1.Get unit-test code
+  $git clone https://github.com/projectacrn/acrn-unit-test.git  # branch: master
```
Comment (junjiemao1) [inline README]:
```suggestion
  $ git clone -b master https://github.com/projectacrn/acrn-unit-test.git
```

If a specific branch is needed, spell it out explicitly in the command.

File: README [hunk 5]
```diff
@@ -1,3 +1,55 @@
...
+
+  2:Enter acrn-unit-test/guest, config and compile. This directory contains sources for ACRN test suite.
+  To create all test images do:
+
+    ./configure --arch=x86_64
```
Comment (junjiemao1) [inline README]:
```suggestion
    $ ./configure --arch=x86_64
```

Be consistent: if we use a prompt in one command, use it everywhere. Same for all command in the rest of this file.

File: README [hunk 6]
```diff
@@ -1,3 +1,55 @@
...
+
+
+  or you can create one special test image:
+
+    ./configure --arch=x86_64
+    make ./x86/msr.flat
+
+  in this directory. Test image is created in ./x86/msr.flat
+  
+# ACRN Hypervisor setup steps, please refer to https://projectacrn.github.io/latest/getting-started/getting-started.html.
```
Comment (junjiemao1) [inline README]:
```suggestion
# For ACRN Hypervisor setup steps, please refer to https://projectacrn.github.io/latest/getting-started/getting-started.html.
```

File: README [hunk 7]
```diff
@@ -1,3 +1,55 @@
...
+  
+
+# Running acrn-unit-test based on ACRN shared scenario.
+  1.Boot ACRN and Service VM, SSH to Service VM.
```
Comment (junjiemao1) [inline README]:
```suggestion
  1. Boot ACRN and Service VM and connect to the Service VM via SSH.
```

Numbered lists generally need a space between the number and the text in Markdown.

File: README [hunk 8]
```diff
@@ -1,3 +1,55 @@
...
+  
+  2.Copy *.flat to Service VM, and run acrn-unit-test.
```
Comment (junjiemao1) [inline README]:
```suggestion
  2. Copy *.flat to Service VM, and run acrn-unit-test.
```

File: README [hunk 9]
```diff
@@ -1,3 +1,55 @@
...
+    $./dm.sh *.flat
+    
+    Example for dm.sh:
+    #!/bin/bash
+    /usr/bin/acrn-dm -A -m 1024M --debugexit -E $1 -s 0:0,hostbridge -s 1:0,lpc -l com1,stdio -U 615db82a-e189-4b4f-8dbb-d321343e4ab3 acrn-vm
+
+  3.Check the test log in HV console.
+    ACRN:\>vm_console 1(VM ID of running *.flat)
```
Comment (junjiemao1) [inline README]: Just curious: does users still have to use the HV console to get the serial log from post-launched VMs?

File: README [hunk 10]
```diff
@@ -1,3 +1,55 @@
...
+
+
+# Add new acrn-unit-test cases
+  1.Currently there is a mix of indentation styles so any changes to existing files should be consistent with the existing style. For new files:
+    - C: please use standard linux-with-tabs, see Linux kernel doc Documentation/process/coding-style.rst
+    - Shell: use TABs for indentation
+  2.Add c file including new cases in acrn-unit-test/guest/x86/., and change lib accordingly.
```
Comment (junjiemao1) [inline README]:
```suggestion
  2.Add c file including new cases in acrn-unit-test/guest/x86/, and change lib accordingly. 
```

**Review by junjiemao1 — CHANGES_REQUESTED** [review]
Here's my two cents.

File: README (hunk 1 above)
Comment (gvancuts) [reply README]: Unless these tests would also be run directly in the Service VM, I would recommend we actually say `in an ACRN User VM`.

File: README (hunk 5 above)
Comment (gvancuts) [reply README]: My preference would be that we do **not** use the command prompt. I prevents people from doing a copy-paste of the instructions.

**Review by gvancuts — CHANGES_REQUESTED** [review]
Thanks for adding details about the repo!

The format of that file looks vaguely like markdown, but isn't proper markdown. I would recommend we actually user ReST here too, just like in the rest of documentation. That way, if we ever want to add this to our main documentation, it's easier to process it through our tooling.

/cc @dbkinder

File: README (hunk 9 above)
Comment (shenfang2019) [reply README]: This is one choice by default. If user does not want to or cannot use console, he must change test code and xml config together.

**Comment by shenfang2019** [comment]
I will change according to all comments. About file format, @terryzouhao and @NanlinXie , do you have any suggestion?

File: README (hunk 1 above)
Comment (shenfang2019) [reply README]: Yes, later acrn-unit-test will be able to work as pre-launched and post-launched VMs, even including service VM.

**Pull request closed without merging**